test/e2e/dra/dra.go

```diff
@@ -411,27 +411,18 @@ var _ = framework.SIGDescribe("node")("DRA", feature.DynamicResourceAllocation,
 			b.create(ctx, claim, pod)
 
 			// Waits for the ResourceClaim to be allocated and the pod to be scheduled.
-			var allocatedResourceClaim *resourceapi.ResourceClaim
-			var scheduledPod *v1.Pod
-
-			gomega.Eventually(ctx, func(ctx context.Context) (*resourceapi.ResourceClaim, error) {
-				var err error
-				allocatedResourceClaim, err = b.f.ClientSet.ResourceV1beta1().ResourceClaims(b.f.Namespace.Name).Get(ctx, claim.Name, metav1.GetOptions{})
-				return allocatedResourceClaim, err
-			}).WithTimeout(f.Timeouts.PodDelete).ShouldNot(gomega.HaveField("Status.Allocation", (*resourceapi.AllocationResult)(nil)))
-
-			gomega.Eventually(ctx, func(ctx context.Context) error {
-				var err error
-				scheduledPod, err = b.f.ClientSet.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
-				if err != nil && scheduledPod.Spec.NodeName != "" {
-					return fmt.Errorf("expected the test pod %s to exist and to be scheduled on a node: %w", pod.Name, err)
-				}
-				return nil
-			}).WithTimeout(f.Timeouts.PodDelete).Should(gomega.BeNil())
```

Comment:
Why did this not ensure that `scheduledPod.Spec.NodeName != ""` when the `gomega.Eventually` succeeded?

Ah, now I see it: if `err != nil`, it exits with nil.

This could be fixed, but using the existing helper is simpler.

/lgtm
/approve